app/services/remember_device_cookie.rb

```diff
@@ -0,0 +1,51 @@
+class RememberDeviceCookie
+  COOKIE_ROLE = 'remember_me'.freeze
+
+  attr_reader :user_id, :created_at
+
+  def initialize(user_id:, created_at:)
+    @user_id = user_id
+    @created_at = created_at
+  end
+
+  def self.from_json(json)
+    parsed_json = JSON.parse(json)
+    check_cookie_role(parsed_json)
+    new(
+      user_id: parsed_json['user_id'],
+      created_at: Time.zone.parse(parsed_json['created_at'])
+    )
+  end
+
+  private_class_method def self.check_cookie_role(parsed_json)
+    role = parsed_json['role']
+    return if role == COOKIE_ROLE
+    raise "RememberDeviceCookie role '#{role}' did not match 'remember_me'"
```

Comment:
Not a blocker for me, but if you're in editing, you might want to interpolate `COOKIE_ROLE` rather than hard-coding the value that it's not matching.

Reply:
Good catch, I agree